src/app/base/components/node/HardwareCard/HardwareCard.test.tsx

```diff
@@ -1,54 +1,42 @@
-import { mount } from "enzyme";
-import { Provider } from "react-redux";
-import { MemoryRouter } from "react-router-dom";
-import configureStore from "redux-mock-store";
+import { screen } from "@testing-library/react";
 
-import HardwareCard from "./HardwareCard";
+import HardwareCard, { Labels as HardwareCardLabels } from "./HardwareCard";
 
 import {
   machineDetails as machineDetailsFactory,
   machineState as machineStateFactory,
   rootState as rootStateFactory,
 } from "testing/factories";
-
-const mockStore = configureStore();
+import { renderWithBrowserRouter } from "testing/utils";
 
 it("renders with system data", () => {
   const machine = machineDetailsFactory({ system_id: "abc123" });
   const state = rootStateFactory({
     machine: machineStateFactory({ items: [machine] }),
   });
-  const store = mockStore(state);
-  const wrapper = mount(
-    <Provider store={store}>
-      <MemoryRouter
-        initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
-      >
-        <HardwareCard node={machine} />
-      </MemoryRouter>
-    </Provider>
-  );
-  expect(wrapper.find("Card")).toMatchSnapshot();
+  renderWithBrowserRouter(<HardwareCard node={machine} />, {
+    route: "/machine/abc123",
+    wrapperProps: { state },
+  });
+
+  // Due to the way a labelled list is rendered, a snapshot unfortunately makes sense here.
+  // You can't check that the information is in the right place since the labels are detached
+  // from the values.
```

Comment:
It's possible you could use [`aria-labelledby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) to associate the labels and values for this.

Reply:
I think a slightly more reliable approach would be to generate a unique id for each list item. This would require moving the label and value into a new component (something like `LabelledListItem`) and then you set up an id like we do here: https://github.com/canonical/maas-ui/blob/main/src/app/base/components/TitledSection/TitledSection.tsx#L71.

So in LabelledList you'd end up with something like:
```
items={items.map((props) => <LabelledListItem {...props} />)}
```

And your new component would be something like:
```
const LabelledListItem = ({ label, value }) => {
  const id = useId();
  return (
    <> 
       <div 
         className="p-list__item-label" 
         id={id} 
       > 
         {label} 
       </div> 
       <div 
         aria-labelledby={id} 
         className="p-list__item-value" 
       > 
         {value} 
       </div> 
     </> 
  )
}
```

Which should be enough for `getByLabelText ` to associate the elements.

Reply:
Ahh I get you yeah. I'm gonna hold off on this just now since I have another PR to merge that also affects these files and I can just smell the merge conflict. Gonna merge that first and then do this one